### ISSUE TITLE
Create a BackgroundJob with a specified Id

### ISSUE THREAD

**Issue by arrkaye** [HangfireIO/Hangfire#1886]
Hi,

Is this possible? I need it to have a predictable Id so that I can cancel it if necessary, before it executes.

Advice is appreciated.

Thanks.

**Comment by Frazi1** [comment]
I don't think you can force Hangfire to create a job with a specific ID. However, when you create a job Hangfire will return the ID back. You can capture it and then use it later to delete the job.

Consider putting your job into `Scheduled` state instead of `Enqueued` to give you some time before it fires. Also, take a look at [filters](https://docs.hangfire.io/en/latest/extensibility/using-job-filters.html) (`IClientFilter` specifically). It gives you the ability to cancel job creation depending on your custom logic.

Can you give more details about your use case? Specifically, why do you need to cancel the job before it executes?
